### PR TITLE
Store full app data in database

### PR DESCRIPTION
**Pull request by vkgnosis**
Part of https://github.com/cowprotocol/services/issues/1465 .

### Test Plan

new e2e test

### CODE REVIEW

File: database/README.md [hunk 1]
```diff
@@ -217,6 +217,7 @@ Column                    | Type                         | Nullable | Details
  class                    | [enum](#orderclass)          | not null | determines which special trade semantics will apply to the execution of this order
  surplus\_fee             | numeric                      | nullable | dynamic fee in sell\_token that gets regularly computed by the protocol for fill-or-kill limit orders, if this is null no surplus\_fee has been computed yet and this order will not be part of our auctions
  surplus\_fee\_timestamp  | timestamptz                  | nullable | last time the backend attempted to update the surplus\_fee for this order, order with too old surplus\_fees will not be put in auctions because it is likely very inaccurate
+ full_app_data            | bytea                        | nullable | The raw bytes of the full app data the order's `app_data` field is a hash of. Null if the backend does not have the full app data. See https://github.com/cowprotocol/services/issues/1465 for more context.
```
Comment (nlordell) [inline database/README.md:220]: Nice! Was going to comment "don't forget to update the DB docs" but that was already done here.

**Review by nlordell — APPROVED** [review]

**Review by MartinquaXD — APPROVED** [review]

File: database/README.md (hunk 1 above)
Comment (MartinquaXD) [inline database/README.md:220]:
```suggestion
 full_app_data            | bytea                        | nullable | The raw bytes of the full app data the order's `app_data` field is a hash of. Null if the backend does not have the full app data. See <https://github.com/cowprotocol/services/issues/1465> for more context.
```

File: crates/e2e/tests/e2e/app_data.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    services.create_order(&order0).await.unwrap();
+
+    // If full app data is set, the hash must match.
+    let app_data = "{}";
+    let app_data_hash = app_data_hash::hash_full_app_data(app_data.as_bytes());
+    let mut order1 = create_order(app_data_hash, Some(app_data.to_string()));
+    services.create_order(&order1).await.unwrap();
+    order1.full_app_data = Some(r#"{"hello":"world"}"#.to_string());
+    let err = services.create_order(&order1).await.unwrap_err();
+    dbg!(err);
```
Comment (cowfee) [inline crates/e2e/tests/e2e/app_data.rs:73]: You're only printing this for inspection? Why not test its contents?

File: crates/model/src/order.rs [hunk 3]
```diff
@@ -660,6 +666,9 @@ pub struct OrderMetadata {
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub onchain_user: Option<H160>,
     pub is_liquidity_order: bool,
+    /// Full app data that `OrderData::app_data` is a hash off. Can be None if
+    /// the backend doesn't know about the full app data.
```
Comment (cowfee) [inline crates/model/src/order.rs:670]: Might be nice to explain what "the backend doesn't know about the full app data" is useful for any where this app data typically is. (It's typically on IPFS, right?)

**Review by cowfee — APPROVED** [review]

**Pull request merged**